src/main/java/de/uni_kl/cs/discodnc/bounds/disco/pwaffine/LeftOverService.java

```diff
@@ -158,15 +158,37 @@ public static ServiceCurve fifoMux(ServiceCurve service_curve, ArrivalCurve arri
      * @return The FIFO service curve
      */
     public static ServiceCurve arbMux(ServiceCurve service_curve, ArrivalCurve arrival_curve) {
-        if (arrival_curve.equals(Curve.getFactory().createZeroArrivals())
-                || service_curve.equals(Curve.getFactory().createZeroDelayInfiniteBurst())) {
-            return service_curve.copy();
-        }
-        if (service_curve.equals(Curve.getFactory().createZeroService())) {
-            return Curve.getFactory().createZeroService();
-        }
-
-        return Curve.getFactory().createServiceCurve(
-                Curve.boundAtXAxis(Curve.sub(service_curve, arrival_curve)));
+    	Pair<Boolean,ServiceCurve> special_cases = computeSpecialValues(service_curve, arrival_curve);
+    	
+    	if(special_cases.getFirst().booleanValue() == true) {
+    		return special_cases.getSecond(); 
+    	} else {
+            return Curve.getFactory().createServiceCurve(Curve.boundAtXAxis(Curve.sub(service_curve, arrival_curve)));
+    	}
+    }
+    
+    private static Pair<Boolean,ServiceCurve> computeSpecialValues(ServiceCurve service_curve, ArrivalCurve arrival_curve) {
+    	ServiceCurve ZERO_SERVICE = Curve.getFactory().createZeroService();
```

Comment:
We should sometime move these constants to a common constant holder, hopefully we do not create new curve instances in the background.

Reply:
We create new curves, we do that a lot, unfortunately. Every operation on a curve will create a new curve instance, too; the same is true for number objects.
This behavior solely exists for historic reasons, i.e., these are the remnants of me not being able to figure out where concurrent modifications of a curve/number objects came from. They happened, did not cause exceptions, "just" caused incorrect results. I sacrificed efficiency to solve this problem by making curves and numbers mostly immutable. But I am of course in favor of slowly and thoroughly improving upon this inefficiency. The common constant holders are a first step into this direction and definitely inspired these lines of code.